### PR TITLE
machines: Fix security and broken packaging in noVNC

### PR DESCRIPTION
**Pull request by stefwalter**
Using unsafe-inline Content-Security-Policy opens up a security in something as wild as noVNC. We simply shouldn't have unsafe-inline or unsafe-eval in our packages.
    
In addition the noVNC component should be installed efficiently, in a single bundle, not having to load 15 javascript files. The bundle should be compressed, with unnecessary dependencies removed.
    
I've added a small custom Webpack loader to perform the task of combining these files.
    
This also allows development on this component to work with pure webpack without automake and friends. We also properly track the dependencies and changes here.

 * [x] #7387

### CODE REVIEW

File: pkg/machines/vnc.js [hunk 1]
```diff
@@ -0,0 +1,194 @@
...
+
+/*
+ * This is a hodgepodge of old-school javascript with hundreds of
+ * interdependencies between files. We concatenate them with our custom
+ * cat.js webpack loader.
+ */
+window.RFB = require("exports?RFB!./cat?noVNC/include/base64.js&noVNC/include/websock.js&noVNC/include/des.js&noVNC/include/keysymdef.js&noVNC/include/keyboard.js&noVNC/include/input.js&noVNC/include/display.js&noVNC/include/rfb.js!noVNC/include/keysym.js");
+
+/*
+    Unresolved issues when trying to bundle noVNC js files:
```
Comment (mareklibra) [inline pkg/machines/vnc.js]: Is this still relevant?

**Comment by martinpitt** [comment]
Fails to build with `SystemError: Could not determine license from cockpit-146.x/node_modules/noVNC`

**Comment by martinpitt** [comment]
Force-pushed to put back `node_modules/noVNC/package.json` into the dist tarball, as that's necessary for determining the license.

File: pkg/machines/manifest.json.in [hunk 2]
```diff
@@ -10,5 +10,5 @@
         "order": 60
      }
   },
-  "content-security-policy": "default-src 'self' 'unsafe-inline' data:"
+  "content-security-policy": "img-src 'self' data:"
```
Comment (martinpitt) [inline pkg/machines/manifest.json.in]: This breaks the "Launch Remote Viewer" button in firefox.

File: pkg/machines/vnc.js (hunk 1 above)
Comment (stefwalter) [inline pkg/machines/vnc.js]: Removed

File: pkg/machines/manifest.json.in (hunk 2 above)
Comment (stefwalter) [inline pkg/machines/manifest.json.in]: Fixed.

**Pull request merged**